File: leetcode/1356_Sort_Integers_by_The_Number_of_1_Bits.py
# URL: https://leetcode.com/problems/sort-integers-by-the-number-of-1-bits/?envType=daily-question&envId=2026-02-25
#
# Problem: 1356. Sort Integers by The Number of 1 Bits
#

# Solution:
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        count = []
        ans = []
        for i in arr:
            count.append((bin(i).count("1"),i))
        count.sort()
        for i in count:
            ans.append(i[1])
        return ans

File: leetcode/1356_Sort_Integers_by_The_Number_of_1_Bits.java
// URL: https://leetcode.com/problems/sort-integers-by-the-number-of-1-bits/description/?envType=daily-question&envId=2026-02-25
//
// Problem: 1356. Sort Integers by The Number of 1 Bits
//

// Solution:
class Solution {
    public int[] sortByBits(int[] arr) {
        int [][] answer = new int [arr.length][2];
        for(int i = 0 ; i < arr.length ; i++){
            answer[i][0] = arr[i];
            answer[i][1] = Integer.bitCount(arr[i]);
            // System.out.println(Integer.bitCount(arr[i]));
        }
        Arrays.sort(answer, (a, b) -> {
            if (a[1] == b[1]) {
                return a[0] - b[0];
            }
            return a[1] - b[1];
        });
        for(int i = 0 ; i < answer.length ; i++){
            arr[i] = answer[i][0];
        }
        System.out.println(Arrays.deepToString(answer));
        return arr;
    }
}